scene/resources/shape_2d.cpp

```diff
@@ -124,7 +124,5 @@ Shape2D::Shape2D(const RID &p_rid) {
 }
 
 Shape2D::~Shape2D() {
-	if (PhysicsServer2D::get_singleton() != nullptr) {
-		PhysicsServer2D::get_singleton()->free(shape);
-	}
+	PhysicsServer2D::get_singleton()->free(shape);
```

Comment:
Don't we still want a guard in here? An error might be preferred over a crash in case something will go wrong.
cc @akien-mga
```suggestion
	ERR_FAIL_NULL(PhysicsServer2D::get_singleton());
	PhysicsServer2D::get_singleton()->free(shape);
```

Reply:
The guard initially didn't exist at all, as most of the code base.

Reply:
Thanks!